src/network/server.cc

```diff
@@ -0,0 +1,238 @@
+    QScopedArrayPointer<char> message(new char[msg_len]);
+    // We don't need it anymore, but we need to get rid of it either way.
+    client_connection->read(reinterpret_cast<char*>(&msg_len), sizeof(msg_len));
+    client_connection->read(&message[0], msg_len);
+    network::Request request;
+    network::Response response;
+    if (!request.ParseFromArray(&message[0], msg_len)) {
+      response.set_ok(false);
+      response.set_error_msg("Failed to decode request.");
+      sendResponse(client_connection, response);
+      return;
+    }
+    // TODO some error handling so that we respond if we fail to process request
+    handleRequest(request, response);
+    sendResponse(client_connection, response);
+  }
+}
+
+void NetworkServer::listChildren(PLocalObject target_object,
+                                 network::Response &resp, bool list_children) {
+  for (auto child : target_object->children()) {
+    network::LocalObject* result = resp.add_results();
+    packObject(child, result, list_children);
+  }
+  resp.set_ok(true);
+}
+
+void NetworkServer::createChunk(PLocalObject target_object, PLocalObject blob,
+                                network::Request &req, network::Response &resp) {
+  PLocalObject created;
+  if (target_object->type() == dbif::FILE_BLOB) {
+     created = ChunkObject::create(target_object, PLocalObject(),
+                                   req.chunk_start(), req.chunk_end(),
+                                   QString::fromStdString(req.chunk_type()),
+                                   QString::fromStdString(req.name()));
+  } else if (target_object->type() == dbif::CHUNK) {
+    created = ChunkObject::create(blob, target_object,
+                                  req.chunk_start(), req.chunk_end(),
+                                  QString::fromStdString(req.chunk_type()),
+                                  QString::fromStdString(req.name()));
+  } else {
+    resp.set_ok(false);
+    resp.set_error_msg("Bad ID provided.");
+    return;
+  }
+  network::LocalObject* result = resp.add_results();
+  packObject(created, result);
+  resp.set_ok(true);
+}
+
+void NetworkServer::deleteObject(PLocalObject target_object,
+                                 PLocalObject root_object,
+                                 network::Response &resp) {
+  if (target_object == root_object) {
+    resp.set_ok(false);
+    resp.set_error_msg("Unable to delete root.");
+    return;
+  }
+  target_object->kill();
+  resp.set_ok(true);
+}
+
+void NetworkServer::handleRequest(network::Request &req, network::Response &resp) {
+  PLocalObject target_object = root_;
+  PLocalObject blob;
+  bool found;
+
+  for (int j = 0; j < req.id_size(); j++) {
+    found = false;
+    for (auto child : target_object->children()) {
+      if (child->id() == req.id(j)) {
+        if (child->type() == dbif::FILE_BLOB) {
+          blob = child;
+        }
+        found = true;
+        target_object = child;
+        break;
+      }
+    }
```

Comment:
Maybe root (or someone) should have a set of all nodes, to avoid bruteforcing our way through the tree?